File: frontend/src/services/customerService.ts
import api from './api'
import type { Customer } from '../types'

export const customerService = {
  async getAll(): Promise<Customer[]> {
    const { data } = await api.get('/api/customers')
    return data
  },

  async search(name: string): Promise<Customer[]> {
    const { data } = await api.get(`/api/customers/search?name=${encodeURIComponent(name)}`)
    return data
  },

  async create(payload: Partial<Customer>): Promise<Customer> {
    const { data } = await api.post('/api/customers', payload)
    return data
  },

  async update(id: number, payload: Partial<Customer>): Promise<Customer> {
    const { data } = await api.put(`/api/customers/${id}`, payload)
    return data
  },

  async delete(id: number): Promise<void> {
    await api.delete(`/api/customers/${id}`)
  },
}

File: backend/src/main/java/com/pos/service/CustomerService.java
package com.pos.service;

import com.pos.dto.CustomerRequest;
import com.pos.entity.Customer;
import com.pos.repository.CustomerRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.math.BigDecimal;
import java.util.List;

@Service
@RequiredArgsConstructor
public class CustomerService {

    private final CustomerRepository customerRepository;

    public List<Customer> getAll() {
        return customerRepository.findByActiveTrue();
    }

    public List<Customer> search(String name) {
        return customerRepository.searchByName(name);
    }

    public Customer getById(Long id) {
        return customerRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Customer not found: " + id));
    }

    @Transactional
    public Customer create(CustomerRequest request) {
        return customerRepository.save(Customer.builder()
                .name(request.getName())
                .email(request.getEmail())
                .phone(request.getPhone())
                .address(request.getAddress())
                .taxId(request.getTaxId())
                .creditBalance(request.getCreditBalance() != null ? request.getCreditBalance() : BigDecimal.ZERO)
                .active(true)
                .build());
    }

    @Transactional
    public Customer update(Long id, CustomerRequest request) {
        Customer customer = getById(id);
        customer.setName(request.getName());
        customer.setEmail(request.getEmail());
        customer.setPhone(request.getPhone());
        customer.setAddress(request.getAddress());
        customer.setTaxId(request.getTaxId());
        if (request.getCreditBalance() != null)
            customer.setCreditBalance(request.getCreditBalance());
        if (request.getActive() != null)
            customer.setActive(request.getActive());
        return customerRepository.save(customer);
    }

    @Transactional
    public void delete(Long id) {
        Customer customer = getById(id);
        customer.setActive(false);
        customerRepository.save(customer);
    }
}
